apps/browser/bookmarks/bookmarkfiltermodel.cpp

```diff
@@ -28,24 +28,53 @@ bool BookmarkFilterModel::filterAcceptsRow(int sourceRow, const QModelIndex &sou
 {
     QModelIndex index = sourceModel()->index(sourceRow, 0, sourceParent);
 
-    if (sourceModel()->data(index, DeclarativeBookmarkModel::UrlRole).toString().trimmed().contains(search(), Qt::CaseInsensitive)
-            || sourceModel()->data(index, DeclarativeBookmarkModel::TitleRole).toString().trimmed().contains(search(), Qt::CaseInsensitive)) {
+    if ((sourceModel()->data(index, DeclarativeBookmarkModel::UrlRole).toString().trimmed().contains(search(), Qt::CaseInsensitive)
+         || sourceModel()->data(index, DeclarativeBookmarkModel::TitleRole).toString().trimmed().contains(search(), Qt::CaseInsensitive))
+         && m_countFilterAccepts < m_maxDisplayedItems) {
```

Comment:
It's likely to be more efficient to move the `m_countFilterAccepts < m_maxDisplayedItems` clause to the start of the condition, and allow the heavier string comparisons to be short-circuited.

You could also use `m_search` directly rather than calling `search()` each time.